src/include/stir/listmode/CListRecordROOT.h

```diff
@@ -165,36 +76,17 @@ class CListRecordROOT : public CListRecord // currently no gating yet
                 raw[1] == dynamic_cast<CListRecordROOT const &>(e2).raw[1];
     }
 
-    CListRecordROOT(const shared_ptr<Scanner>& scanner_sptr) :
-        event_data(scanner_sptr)
-    {}
+    CListRecordROOT(const shared_ptr<Scanner>& scanner_sptr);
 
     virtual Succeeded init_from_data( const int& ring1,
                                       const int& ring2,
                                       const int& crystal1,
                                       const int& crystal2,
                                       double time1, double time2,
-                                      const int& event1, const int& event2)
-    {
-        /// \warning ROOT data are time and event at the same time.
-
-        this->event_data.init_from_data(ring1, ring2,
-                                        crystal1, crystal2);
-
-        this->time_data.init_from_data(
-                    time1,time2);
-
-        // We can make a singature raw based on the two events IDs.
-        // It is pretty unique.
-        raw[0] = event1;
-        raw[1] = event2;
-
-        return Succeeded::yes;
-    }
-
+                                      const int& event1, const int& event2);
 private:
-    CListEventROOT  event_data;
-    CListTimeROOT   time_data;
+    CListEventROOT*  event_data;
```

Comment:
this looks terrible. a raw pointer? why?

Reply:
Sure :S is was only temporal